docs/src/content/en/docs/agents/overview.mdx

```diff
@@ -54,264 +78,237 @@ export const mastra = new Mastra({
 });
 ```
 
-## 2. Generating and streaming text
+## Referencing an agent
+
+You can call agents from workflow steps, tools, the Mastra Client, or the command line. Get a reference by calling `.getAgent()` on your `mastra` or `mastraClient` instance, depending on your setup:
+
+```typescript showLineNumbers copy
+const testAgent = mastra.getAgent("testAgent");
+```
+
+> See [Calling agents](../../examples/agents/calling-agents.mdx) for more information.
+
+## Generating responses
+
+Use `.generate()` to get a response from an agent. Pass a single string for simple prompts, an array of strings when providing multiple pieces of context, or an array of message objects with `role` and `content` for precise control over roles and conversational flows.
+
+> See [.generate()](../../reference/agents/generate.mdx) for more information.
 
 ### Generating text
 
-Use the `.generate()` method to have your agent produce text responses:
+Call `.generate()` with an array of message objects that include `role` and `content`:
 
 ```typescript showLineNumbers copy
 const response = await testAgent.generate([
-  { role: "user", content: "Hello, how can you assist me today?" },
+  { role: "user", content: "Help me organize my day" },
+  { role: "user", content: "My day starts at 9am and finishes at 5.30pm" },
+  { role: "user", content: "I take lunch between 12:30 and 13:30" },
+  { role: "user", content: "I have meetings Monday to Friday between 10:30 and 11:30" }
 ]);
```

Comment:
This isn't a common pattern for how people use this.

We should recommend people pass a string here and call out that they can add additional messages if they want to build up a conversation history (user, assistant, user, assistant, etc) but they only need to do that if they're managing memory themselves without mastra memory.



```suggestion
const response = await testAgent.generate(
  `Help me organize my day. My day starts at 9am and finishes at 5.30pm
  I take lunch between 12:30 and 13:30
  I have meetings Monday to Friday between 10:30 and 11:30`
);
```

Reply:
for passing an array it'd look more like:

```ts
const response = await testAgent.generate([
  { role: "user", content: "Help me organize my day" },
  { role: "assistant", ... },
  { role: "tool", ... },
  { role: "assistant", ... }
]);
```

Reply:
but I think array here is overcomplicating things

Reply:
I'd agree, and would normally just show a string message, but because our API references aren't quite up to scratch there's no where else to explain the messages object.